subsys/settings/src/settings_zms.c

```diff
@@ -3,6 +3,8 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#define _POSIX_C_SOURCE 200809L /* for strnlen() */
```

Comment:
Add `#undef _POSIX_C_SOURCE` or `#ifndef` before to make sure this compiles if a user sets its own POSIX version.